### PR TITLE
Adds charts in "charts\" directory to dependencies

### PR DESCRIPTION
**Pull request by sushilkm**
Dependencies should be processed even if they are not added to requirements.yaml
Fixes https://github.com/kubernetes/helm/issues/2599

### CODE REVIEW

File: pkg/chartutil/requirements.go [hunk 1]
```diff
@@ -258,6 +258,24 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 	}
 
 	var chartDependencies []*chart.Chart
+	// If any dependency is not a part of requirements.yaml
+	// then this should be added to chartDependencies.
+	// However, if the dependency is already specified in requirements.yaml
+	// we should not add it, as it would be anyways processed from requirements.yaml
+
```
Comment (jascott1) [inline pkg/chartutil/requirements.go]: seems verbose, just my imo but consider shortening to something like
"// add dependencies from /charts not already present in requirements.yaml"

Comment (technosophos) [inline pkg/chartutil/requirements.go]: I kinda prefer the more verbose description, since I have the long-term memory of a sea slug. :-) But I'm okay with whatever.

Comment (technosophos) [inline pkg/chartutil/requirements.go]: ROTFL. After writing that, I googled "Sea Slug Long Term Memory"... and discovered that they actually do memory research on sea slugs. https://www.livescience.com/17683-sea-slug-memory.html

**Comment by technosophos** [comment]
We should catalog some test cases we need to verify. Off the top of my head, it's:

- subcharts, no requirements.yaml
- subcharts, all listed in requirements.yaml
- subcharts, some in requirements, some not.

Am I missing and edge cases or other pertinent scenarios? Do we need to do testing based on whether or not charts are enabled/disabled by a particular requirements file?

**Comment by chancez** [comment]
Might be good to validate that chart's containing subcharts, that have subcharts, too. Not something I use, but some limited testing of various nesting cases is probably good.

File: pkg/chartutil/requirements.go (hunk 1 above)
Comment (jascott1) [inline pkg/chartutil/requirements.go]: Part of the new code review process? "They tested the model by submitting the animals to a series of training sessions, involving electric shocks"

**Comment by jascott1** [comment]
@sushilkm You might want to checkout the testdata/subpop chart as it already has a bunch of nested dependencies. You could add some charts that aren't in a requirements.yaml.

**Comment by sushilkm** [comment]
@jascott1 I think the only case `testdata/subpop` can help is for trying all subcharts specified in `requirements.yaml`, or is there anything else you want me to test with `subpop`

**Comment by jascott1** [comment]
@sushilkm nothing specific, just thought it might be useful to modify and test with since it already has deep nested charts. Sounds like you have it covered :)

**Comment by thomastaylor312** [comment]
Marked this as small since most of the code is just test charts 😄

**Comment by technosophos** [comment]
Are we ready to re-review and merge this?

I can break out the low-voltage system and shock some sea slugs @jascott1 ;-)

**Comment by scottrigby** [comment]
@sushilkm Great work. Also love the lyrical test data. :shipit:

**Comment by jascott1** [comment]
@technosophos I think it needs re-review now as it seems ready. Thats correct @sushilkm?

**Comment by sushilkm** [comment]
Yup waiting for reviews on this one

**Comment by technosophos** [comment]
@thomastaylor312 When you marked it "in process", were you claiming this one? If not, I can do a re-review.

**Comment by thomastaylor312** [comment]
@technosophos Sorry no, I was just fixing the label because people were already reviewing it.

**Comment by technosophos** [comment]
Okay. Gonna try to get this tested and committed before the k8s 1.7 PR is merged.

**Review by technosophos — APPROVED** [review]
pulled and manually ran tests.
